### ISSUE TITLE
Seems to crash with KIS

### ISSUE THREAD

**Issue by warmist** [mockingbirdnest/Principia#1549]
Crashing when trying to add a part to spacecraft.
log:
`Log file created at: 2017/09/01 15:43:18
Running on machine: DESKTOP-P4285VI
Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line] msg
    @   00007FFCFFEA7799  	principia__RenderedVesselTrajectory [0x00007FFCFFEA7798+218356]
    @   00007FFCFFE1735F  	principia__FlightPlanRemoveLast [0x00007FFCFFE1735E+571018]
    @   00007FFCFFE354EB  	principia__FlightPlanRemoveLast [0x00007FFCFFE354EA+694294]
    @   00007FFCFFD1AEE8  	principia__IteratorAtEnd [0x00007FFCFFD1AEE7+20463]
    @   00007FFCFFD3C620  	principia__FreeVesselsAndPartsAndCollectPileUps [0x00007FFCFFD3C61F+107]
    @   000000001FB7EBDC  	(No symbol) [0x000000001FB7EBDB]
F0901 15:43:18.329875 10436 vessel.cpp:127] Check failed: !parts_.empty() `

Edit: testing it further, it only crashes when you drop an item.

**Comment by eggrobin** [comment]
Thanks for the report.
As mentioned [in the FAQ](https://github.com/mockingbirdnest/Principia/wiki/Installing,-reporting-bugs,-and-frequently-asked-questions#windows-dialog-box-or-sigabrt), we need the *full* INFO log to investigate crashes, so please provide that. This is because it starts with additional information not contained in the WARNING, ERROR, or FATAL logs; without it the stack trace is useless.

**Comment by warmist** [comment]
Sorry. Here is more data: https://gist.github.com/warmist/eebd253a85efea9db4b7b9ad9518e200

**Comment by pleroy** [comment]
Thanks for the additional information.

For our reference:

[`ksp_plugin/vessel.cpp:127`](https://github.com/mockingbirdnest/Principia/blob/c9d0610117bbfaa4633fa3f2d5b72b99547f7151/ksp_plugin/vessel.cpp#L127)
[`ksp_plugin/plugin.cpp:471`](https://github.com/mockingbirdnest/Principia/blob/c9d0610117bbfaa4633fa3f2d5b72b99547f7151/ksp_plugin/plugin.cpp#L471)
[`ksp_plugin/interface.cpp:356`](https://github.com/mockingbirdnest/Principia/blob/c9d0610117bbfaa4633fa3f2d5b72b99547f7151/ksp_plugin/interface.cpp#L356)

**Comment by pleroy** [comment]
From the look of it, KIS creates a vessel to represent the part:
```
I0901 16:47:21.103809  6832 plugin.cpp:361] Inserted loaded vessel SP-L 1x6 Photovoltaic Panels (e1e64236-6d2c-44f3-9ed8-db30e3a20aee)
```
I suspect that it will later try to dock it with the main vessel.  The problem is that this part-pretending-to-be-a-vessel doesn't contain a part.  Compare the above trace with what happens during an EVA:
```
I0901 16:11:57.701596  6832 plugin.cpp:361] Inserted loaded vessel Bill Kerman (bd5865d7-0e82-4c9e-946b-0a47b8e7d084)
I0901 16:11:57.701596  6832 vessel.cpp:81] Adding part kerbalEVA (Bill Kerman) (5A1655BA) to vessel Bill Kerman (bd5865d7-0e82-4c9e-946b-0a47b8e7d084)
```
Now Principia expects all the physical attributes (mass, position) to be attached to parts, not vessels.  A vessel without a part cannot be integrated.  Hence the failure.

The bottom line is that KIS violates the invariants that the game normally enforces and that Principia depends upon.  Unclear what is the right course of action here: it's already quite hard to make KSP happy, I'm not sure we want to deal with other mods doing weird things.

**Comment by warmist** [comment]
I guess this should be forwarded to KIS then...
Thanks for all the work :)

**Comment by pleroy** [comment]
Fixed in KIS, thanks to @ihsoft.
